### PR TITLE
[CIS-547] Add support for giphy attachments

### PR DESCRIPTION
**Pull request by ManWithBear**
Add support for gif images via new 3rd party dependency [`SwiftyGif`](https://github.com/kirualex/SwiftyGif) (already used in v2).

https://user-images.githubusercontent.com/6978940/103332376-f6f1df00-4a69-11eb-81bc-7076dfe1fccc.mov

### CODE REVIEW

**Review by VojtaStavik — APPROVED** [review]
Looks great, thank you! I thought Nuke would be able to show gif images too. Apparently not :/

File: Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import Nuke
+import StreamChat
+import SwiftyGif
+import UIKit
+
+open class ChatMessageGiphyView<ExtraData: ExtraDataTypes>: View, UIConfigProvider {
+    public var content: _ChatMessageAttachment<ExtraData>? {
+        didSet {
+            guard oldValue?.imageURL != content?.imageURL else { return }
```
Comment (evsaev) [inline Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift:13]: What if old and new imageURLs match but previous downloading has failed? 🤔
As an option we can check if image view is displaying gif and if not - download even though urls are the same.

File: Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageAttachmentsView.swift [hunk 2]
```diff
@@ -69,6 +85,7 @@ open class ChatMessageAttachmentsView<ExtraData: ExtraDataTypes>: View, UIConfig
         ]
         layoutConstraints[[.images, .interactiveAttachment]] = layoutConstraints[.interactiveAttachment]
         layoutConstraints[[.files, .interactiveAttachment]] = layoutConstraints[.interactiveAttachment]
+        layoutConstraints[[.giphy, .interactiveAttachment]] = layoutConstraints[.interactiveAttachment]
```
Comment (evsaev) [inline Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageAttachmentsView.swift:88]: Shouldn't we also provide a layout for `[.giphy, .images]` options?

File: Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageAttachmentsView.swift [hunk 3]
```diff
@@ -121,8 +141,10 @@ open class ChatMessageAttachmentsView<ExtraData: ExtraDataTypes>: View, UIConfig
         for attachment in content?.attachments ?? [] {
             if !attachment.actions.isEmpty {
                 options.insert(.interactiveAttachment)
-            } else if attachment.isImageOrGIF {
```
Comment (evsaev) [inline Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageAttachmentsView.swift:124]: Do we any place where `isImageOrGIF` is used? Looks it's not relevant anymore

File: Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift [hunk 4]
```diff
@@ -0,0 +1,70 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Nuke
+import StreamChat
+import SwiftyGif
+import UIKit
+
+open class ChatMessageGiphyView<ExtraData: ExtraDataTypes>: View, UIConfigProvider {
```
Comment (evsaev) [inline Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift:10]: nit: in the design the gif displaying component has `GIPHY` label. Would be nice to have it, but that's not super important

File: Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageAttachmentsView.swift (hunk 2 above)
Comment (ManWithBear) [reply Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageAttachmentsView.swift:88]: Is user able to put giphy and images in one message?
I think no and in this case it is good for now. 
As we previously speak, we will rework all `LayoutOption` things soon (they not scale well). Will be action point

File: Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageAttachmentsView.swift (hunk 3 above)
Comment (ManWithBear) [reply Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageAttachmentsView.swift:124]: Yes, it's deleted in 2c7a17b

File: Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift (hunk 4 above)
Comment (ManWithBear) [reply Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift:10]: My bad, totally missed it.

File: Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift (hunk 1 above)
Comment (ManWithBear) [reply Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift]: d1e2405

File: Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift (hunk 4 above)
Comment (ManWithBear) [reply Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift:10]: c230616

**Review by evsaev — APPROVED** [review]

**Pull request merged**